lib/corm/model.rb

```diff
@@ -138,6 +140,43 @@ def self.count
       execute(statements['count']).first['count'].to_i
     end
 
+    ##
+    # Find by keys.
+    # This `find` methods wants to be as flexible as possible.
+    #
+    # Unless a block is given, it returns an `Enumerator`, otherwise it yields
+    # to the block an instance of the found Cassandra entries.
+    #
+    # If no keys as passed as parameter, the methods returns (an Enumerator for)
+    # all the results in the table.
+    #
+    # If one or more keys are given, the **order matters**; since this methods
+    # will craft a CSQL query, the first key **must** be the `partition key`,
+    # while the followings are intended as the `clustering keys`... that can be
+    # more than one.
+    #
+    # If the keys passed as parameter are more than the defined by the table the
+    # query is not valid, it cannot be executed and an error is raised.
+    def self.find(*key_values, &block)
+      return to_enum(:find, *key_values) unless block_given?
+
+      field_names = []
+      Array(key_values).each_with_index do |value, index|
+        field_names << "#{primary_key.flatten[index]} = ?"
+      end
+
+      statement_find_key = "find_#{Digest::MD5.hexdigest(field_names.inspect)}"
```

Comment:
I'm sorry, am I missing something? **values** are not taken in consideration when generating prepared statements, as they are added when the statements is used via bind.

I agree that avoiding hashes makes the whole thing more explicit, which is good.
So given a partition key `[id, id2]` and clustering keys `[c1, c2]` we end up with three prepared statements:

- `find_id_id2`
- `find_id_id2_c1`
- `find_id_id2_c1_c2`